### PR TITLE
Supports multiple accounts with the same Passkey

### PR DESCRIPTION
**Pull request by zkJoaquin**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: system-contracts/contracts/PasskeyBinder.sol [hunk 1]
```diff
@@ -65,50 +59,54 @@ contract PasskeyBinder is IPasskeyBinder {
...
-        if (publicKeyOwner == address(0)) revert KeyDoesNotExist(_credentialIdHash);
-        if (publicKeyOwner != sender) revert DoesNotOwner(_credentialIdHash);
+        bool ownershipOfRight = credentialIdHashSet.contains(_credentialIdHash);
+        if (!ownershipOfRight) revert DoesNotOwnershipOfRights(_credentialIdHash);
 
-        uint256 x = authorizedKey.publicKey.x;
-        uint256 y = authorizedKey.publicKey.y;
-
-        delete authorizedKeys[_credentialIdHash];
         accountToCredentialIdHashSet[sender].remove(_credentialIdHash);
```
Comment (zkbenny) [inline system-contracts/contracts/PasskeyBinder.sol:83]:
```solidity
if(!accountToCredentialIdHashSet[sender].remove(_credentialIdHash)) {
 revert DoesNotOwnershipOfRights(_credentialIdHash);
}
```

File: system-contracts/contracts/PasskeyBinder.sol [hunk 2]
```diff
@@ -65,50 +59,54 @@ contract PasskeyBinder is IPasskeyBinder {
...
-
-        owner = authorizedKey.owner;
-        x = authorizedKey.publicKey.x;
-        y = authorizedKey.publicKey.y;
+    function getAuthorizedKey(
+        address _account,
+        bytes32 _credentialIdHash
+    ) external view returns (uint256 x, uint256 y) {
+        EnumerableSet.Bytes32Set storage credentialIdHashSet = accountToCredentialIdHashSet[_account];
+        P256PublicKey memory publicKey = authorizedKeys[_credentialIdHash].publicKey;
```
Comment (zkbenny) [inline system-contracts/contracts/PasskeyBinder.sol:100]: mem copy publicKey when ownershipOfRight is true

File: system-contracts/contracts/PasskeyBinder.sol [hunk 3]
```diff
@@ -65,50 +59,54 @@ contract PasskeyBinder is IPasskeyBinder {
 
         if (_credentialIdHash == bytes32(0)) revert InvalidCredentialIdHash();
 
-        AuthorizedKey storage authorizedKey = authorizedKeys[_credentialIdHash];
-
-        if (authorizedKey.owner != address(0)) revert KeyAlreadyExists(_credentialIdHash);
+        address authorizedKeyOwner = authorizedKeys[_credentialIdHash].owner;
+        if (authorizedKeyOwner == address(0)) {
+            authorizedKeys[_credentialIdHash] = AuthorizedKey(sender, P256PublicKey(_x, _y));
```
Comment (zkbenny) [inline system-contracts/contracts/PasskeyBinder.sol:64]: how to prevent fake (x, y) attack?

Comment (zkJoaquin) [reply system-contracts/contracts/PasskeyBinder.sol:64]: (x, y) is obtained from the Passkey generated by the device. When a fake (x, y) is bound, the signature check fails.

Comment (zkJoaquin) [reply system-contracts/contracts/PasskeyBinder.sol]: Front-Running, there is a possibility of contaminating other users' passwords. But it has been fixed.

**Review by zkbenny — APPROVED** [review]

**Pull request merged**